front/src/app/components/UploadFile.tsx

```diff
@@ -1,45 +1,132 @@
+import { fr } from "@codegouvfr/react-dsfr";
+import { Alert } from "@codegouvfr/react-dsfr/Alert";
 import * as React from "react";
-import { File } from "react-design-system";
+import { CopyButton, File } from "react-design-system";
 import { AbsoluteUrl } from "shared";
-import { domElementIds } from "shared";
+import { useAppSelector } from "src/app/hooks/reduxHooks";
 import { outOfReduxDependencies } from "src/config/dependencies";
+import { authSelectors } from "src/core-logic/domain/auth/auth.selectors";
 
 interface UploadLogoProps {
   label: string;
   hint?: React.ReactNode;
   maxSize_Mo: number;
-  setFileUrl: (fileUrl: AbsoluteUrl) => void;
+  setUploadedFileUrl?: (fileUrl: AbsoluteUrl) => void;
+  initialFileUrl?: AbsoluteUrl;
+  id: string;
+  shouldDisplayFeedback: boolean;
 }
 
 export const UploadFile = ({
   maxSize_Mo,
-  setFileUrl,
+  setUploadedFileUrl,
   label,
   hint,
+  id,
+  shouldDisplayFeedback,
+  initialFileUrl,
 }: UploadLogoProps) => {
+  const copyButtonLabel = "Copier l'URL du fichier";
   const [error, setError] = React.useState<string>();
+  const [uploadError, setUploadError] = React.useState<string>();
+  const [file, setFile] = React.useState<File | null>(null);
+  const [fileUrl, setFileUrl] = React.useState<AbsoluteUrl | null>(null);
+  const [fileBase64, setFileBase64] = React.useState<
+    string | ArrayBuffer | null
+  >(null);
+  const connectedUserJwt = useAppSelector(authSelectors.inclusionConnectToken);
+
+  if (!connectedUserJwt) return null;
+
+  const toBase64 = (file: File) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onload = () => setFileBase64(reader.result);
+    reader.onerror = () => setFileBase64(null);
+  };
+
+  const setUploadedFile = (file: File, fileUrl: AbsoluteUrl) => {
+    setFile(file);
+    toBase64(file);
+    setFileUrl(fileUrl);
+    setUploadedFileUrl ? setUploadedFileUrl(fileUrl) : null;
+  };
+
   return (
-    <File
-      onChange={async (e) => {
-        let file = null;
-        if (e.target.files?.length) {
-          file = e.target.files[0];
-        }
-        if (file && file.size > 1_000_000 * maxSize_Mo) {
-          setError(`Le fichier ne peut pas faire plus de ${maxSize_Mo} Mo`);
-          return;
-        }
-        setError(undefined);
-        if (!file) return;
+    <>
+      <File
+        onChange={async (e) => {
+          let file = null;
+          if (e.target.files?.length) {
+            file = e.target.files[0];
+          }
```

Comment:
Eviter le patern let

```suggestion
const file =  e.target.files[0] ??  null
```